src/main/kotlin/ecommerce/model/CartItem.kt

```diff
@@ -0,0 +1,37 @@
+package ecommerce.model
+
+import ecommerce.dto.CartItemResponse
+import jakarta.persistence.Entity
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import org.hibernate.annotations.CreationTimestamp
+import java.time.LocalDateTime
+
+@Entity
+data class CartItem(
+    @ManyToOne
+    @JoinColumn(name = "cart_id")
+    var cart: Cart,
+    @ManyToOne
+    @JoinColumn(name = "product_id")
+    val product: Product,
+    var quantity: Int,
+    @CreationTimestamp
+    var createdAt: LocalDateTime = LocalDateTime.now(),
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    val id: Long = 0L,
```

Comment:
```suggestion
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    @ManyToOne
    @JoinColumn(name = "cart_id")
    var cart: Cart,

    @ManyToOne
    @JoinColumn(name = "product_id")
    val product: Product,

    var quantity: Int,

    @CreationTimestamp
    var createdAt: LocalDateTime = LocalDateTime.now(),
```

I personally prefer keeping `id` field on the very top, since that (to me) is the most important value when it comes to entity.

Also, when there's multiple annotations added, I like to keep some line breaks in between fields, as it helps increase the readability. (this is also just my opinion 😂 )

Reply:
😄 I actually had the id at the top initially as well, mostly because I was used to that style from working with Java.
But during the sessions with Json and Justin, they suggested keeping it at the bottom so we don’t have to include it in the constructor.

I found this especially convenient for creating entities in tests, since we can omit the id entirely when building test objects. So for me, it’s more practical and less boilerplate.

Reply:
Sounds great to me! 👍